### PR TITLE
feat!: make DhtHeader field non-malleable

### PR DESCRIPTION
**Pull request by philipr-za**
Description
---
This PR includes the relevant DhtHeader fields in the commitment of a Dht Message MAC signature so that they cannot be changed while in route.

The header fields included in the commitment are:
- Major version
- Minor version
- Destination
- Message type
- Message flags
- Expiry time (if exists)
- Ephemeral public key (if exists)

How Has This Been Tested?
---
Rust integration test provided and existing tests updated.

BREAKING CHANGE: Previous message signatures will not be valid

### CODE REVIEW

File: comms/dht/src/inbound/decryption.rs [hunk 1]
```diff
@@ -193,8 +193,17 @@ where S: Service<DecryptedDhtMessage, Response = (), Error = PipelineError>
     ) -> Result<DecryptedDhtMessage, DecryptionError> {
         let dht_header = &message.dht_header;
 
+        let mut header_mac_bytes = dht_header.major.to_le_bytes().to_vec();
```
Comment (delta1) [inline comms/dht/src/inbound/decryption.rs]: minor perf: is it worth allocating this vec with capacity before? Maybe not necessary without a profile but there would be a few reallocations.

File: comms/dht/src/test_utils/makers.rs [hunk 2]
```diff
@@ -75,14 +75,28 @@ pub fn make_dht_header(
     flags: DhtMessageFlags,
     include_origin: bool,
     trace: MessageTag,
+    include_destination: bool,
```
Comment (delta1) [inline comms/dht/src/test_utils/makers.rs:79]: lil clippy warning here

File: comms/dht/tests/dht.rs [hunk 3]
```diff
@@ -629,6 +630,99 @@ async fn dht_propagate_message_contents_not_malleable_ban() {
...
+    #[derive(Clone, PartialEq, ::prost::Message)]
+    struct Person {
+        #[prost(string, tag = "1")]
+        name: String,
+        #[prost(uint32, tag = "2")]
+        age: u32,
+    }
+
+    let out_msg = Person {
+        name: "John Conway".into(),
```
Comment (delta1) [inline comms/dht/tests/dht.rs:665]: RIP 🙏

**Review by delta1 — DISMISSED** [review]
Looking great, happy to approve with clippy passing

File: comms/dht/src/inbound/decryption.rs (hunk 1 above)
Comment (philipr-za) [reply comms/dht/src/inbound/decryption.rs]: I guess you might as well

File: comms/dht/tests/dht.rs (hunk 3 above)
Comment (philipr-za) [reply comms/dht/tests/dht.rs:665]: damnit, I forget to check the tests locally

File: comms/dht/src/test_utils/makers.rs (hunk 2 above)
Comment (philipr-za) [reply comms/dht/src/test_utils/makers.rs:79]: damnit I forget to check the tests locally

**Review by delta1 — DISMISSED** [review]
LGTM

**Review by stringhandler — COMMENTED** [review]
Happy to merge this once a WIREMODE update is included

**Comment by philipr-za** [comment]
> Happy to merge this once a WIREMODE update is included

Ahh yes you are correct I should have incremented the DHT_MAJOR_VERSION and have updated that.

**Review by sdbondi — APPROVED** [review]
utAck - obviously would be nice to DRY up the challenge construction

**Pull request closed without merging**